ranger-core/src/main/java/io/appform/ranger/core/serviceprovider/ServiceProvider.java

```diff
@@ -69,10 +76,30 @@ private void handleHealthUpdate(HealthcheckResult result) {
             log.debug("No update to health state of node. Skipping data source update.");
             return;
         }
+        setStartupTimeIfAbsent(result);
+        if (null != weightSupplier && weightSupplier.isEnabled() && isBetweenZeroAndOne(
+                weightSupplier.getSupplier().get())) {
+            serviceNode.setWeight(weightSupplier.getSupplier().get());
```

Comment:
This is wrong. the supplier needs to be called _only_ once and null needs to be checked in response and defaulted to something. The supplier might be  a time or invocation count based ramp-up function and calling it multiple times can possibly yield different values